images/replication-merger/start.py

```diff
@@ -0,0 +1,374 @@
+    log.info(f'fetching file from {url}')
+    result = requests.get(url)
+    return (result.ok, result.text)
+
+def extract_positive_number(file_content, key):
+    try:
+        value = extract_value(content=file_content, key=key, is_int_value=True, delimiter='=', 
+                                throw_not_found=False, default_value=0)
+        if value < 0:
+            raise ValueError()
+        return value
+    except:
+        log_and_exit(f'{key} must be a positive integer')
+
+def time_zone_to_utc(datetime):
+    time_zone_dt = pytz.timezone(OSMOSIS_TIME_ZONE).localize(datetime)
+    return time_zone_dt.astimezone(pytz.utc)
+
+def extract_value(content, key, is_int_value=True, delimiter='=', throw_not_found=False, default_value=-1):
+    """
+    Extracts a key from content, value can be an integer or string
+    Args:
+        content (str): the full given text content
+        key (str): the wanted key to be searched in the given content
+        is_int_value (bool): determines if the key's value is int or string, which effects the search and parsing of the value
+        delimiter (str): the separator between the key and it's value to be splitted by
+        throw_not_found (bool): a flag determines if to raise a LookupError
+        default_value (any): the value returned upon not finding the key in the content while not throwing an error
+    Raises:
+        LookupError: throw_not_found is true and key could not be found
+        ValueError: is_int_value is true and the value is not a positive integer or an error while parsing the value
+    Returns:
+        (int|str): the extracted value
+    """
+    if is_int_value:
+        match = re.search(fr'{key}=\d+', content)
+    else:
+        match = re.search(fr'{key}=\S+', content)
+    if not match:
+        if throw_not_found:
+            raise LookupError(f'"{key}=" is not present in the given content')
+        else:
+            return default_value
+    
+    value = match.group(0).split(delimiter)[1]
+    try:
+        return int(value) if is_int_value else str(value)
+    except ValueError:
+        raise ValueError('an error accourd while extraction.')
+
+def parse_time_unit_config(config_content, fetch_from_file_system = False):
+    """
+    Extracts the wanted values from a time unit config
+    Args:
+        config_content (str): the config text
+        fetch_from_file_system: (bool): flag
+    Returns:
+        Dictionary: key value pair extracted from config
+    """
+    interval_length = extract_positive_number(config_content, MERGE_CONFIG_MAP['INTERVAL_LENGTH'])
+    max_interval = extract_positive_number(config_content, MERGE_CONFIG_MAP['MAX_INTERVAL'])
+    if fetch_from_file_system:
+        sub_time_unit_data_dir = extract_value(config_content, MERGE_CONFIG_MAP['BASE_URL'], False, MERGE_CONF_BASE_URL_HEADER)
+    else:
+        sub_time_unit_data_dir = extract_value(config_content, MERGE_CONFIG_MAP['BASE_URL'], False, REPLICATION_URL + '/')
```

Comment:
same here, please use named arguements

Reply:
done.

Reply:
have you pushed the code? I don't see that it is fixed